Resources/Catalog/FileCatalog.py

```diff
@@ -123,27 +161,27 @@ def w_execute( self, *parms, **kws ):
         else:
           # Otherwise we keep the failed catalogs so we can update their state later
           failedCatalogs.append( ( catalogName, res['Message'] ) )
-      else:
-        if 'Failed' in res['Value']:
-          lfnsFlag = True
-          for lfn, message in res['Value']['Failed'].items():
-            # Save the error message for the failed operations
-            failed.setdefault( lfn, {} )[catalogName] = message
-            if master:
-              # If this is the master catalog then we should not attempt the operation on other catalogs
-              fileInfo.pop( lfn, None )
-          for lfn, result in res['Value']['Successful'].items():
-            # Save the result return for each file for the successful operations
-            successful.setdefault( lfn, {} )[catalogName] = result
+      for lfn, message in res['Value']['Failed'].items():
```

Comment:
This will make an exception if the failure is on a non master catalog. You don't have res['Value'] then. 
You need to keep the 'else' clause that was here before